client/state/sites/domains/schema.js

```diff
@@ -0,0 +1,37 @@
+export const itemsSchema = {
+	type: 'array',
```

Comment:
Right, my confusion was that I wasn't aware of it's `{ 'items': ... }` schema:

The site.domains subtree must to be:

```js
domains: {
  items: {
    site_Id_0: [ domain_0, domain_1, ..., domain_n ] ],
    site_Id_1: [ domain_0, domain_1, ..., domain_n ] ],
    // ...
    site_Id_n: [ domain_0, domain_1, ..., domain_n ] ]
  },

  requesting: {},

  errors: {}
}
```